app/src/main/java/com/example/plantsapp/presentation/ui/authentication/AuthViewModel.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.plantsapp.presentation.ui.authentication
+
+import android.content.Intent
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.example.plantsapp.R
+import com.example.plantsapp.presentation.core.Event
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.common.api.ApiException
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.GoogleAuthProvider
+import dagger.hilt.android.lifecycle.HiltViewModel
+import timber.log.Timber
+import javax.inject.Inject
+
+@HiltViewModel
+class AuthViewModel @Inject constructor(
+    private val auth: FirebaseAuth
+) : ViewModel() {
+
+    private val _navigateToSplash: MutableLiveData<Event<Unit>> = MutableLiveData()
+    val navigateToSplash: LiveData<Event<Unit>> get() = _navigateToSplash
+    private val _signingError: MutableLiveData<Int> = MutableLiveData()
+    val signingError: LiveData<Int> get() = _signingError
+
+    fun onSignInResult(intent: Intent) {
+        val task = GoogleSignIn.getSignedInAccountFromIntent(intent)
+        try {
+            val account = task.getResult(ApiException::class.java)
+            signInFirebase(account.idToken!!)
+        } catch (e: ApiException) {
+            Timber.e(e)
+            _signingError.value = R.string.error_unable_to_sign_in
+        }
+    }
```

Comment:
Okay, it is time to move this to something else 
1. Intent is huge of android , so we need to somehow hide it from our view model
2. How about create User domain model that will store small amount of data needed for us
3. Also move this thing to AuthUseCase - so you will just pass some input and if use case works that means that user successfully logged-in (make this method suspend + also for now you can leave Intent as input - we will think on it later)